File: JavaScript/encryptor.js
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

function generateKey() {
    return crypto.randomBytes(32);
}

function padData(data) {
    const blockSize = 16;
    const paddingLength = blockSize - (data.length % blockSize);
    const padding = Buffer.alloc(paddingLength, paddingLength);
    return Buffer.concat([data, padding]);
}

function encryptFile(filePath, key) {
    const iv = crypto.randomBytes(16);
    const cipher = crypto.createCipheriv('aes-256-cbc', key, iv);

    const data = fs.readFileSync(filePath);
    const paddedData = padData(data);
    const encryptedData = Buffer.concat([cipher.update(paddedData), cipher.final()]);

    fs.writeFileSync(filePath, Buffer.concat([iv, encryptedData]));
}

function processFolder(folderPath, key) {
    fs.readdirSync(folderPath).forEach(file => {
        const filePath = path.join(folderPath, file);
        if (fs.statSync(filePath).isFile()) {
            encryptFile(filePath, key);
        }
    });
}

function main() {
    const folderPath = "E:/Test/Folder";
    const parentPath = path.dirname(folderPath);
    const keyFilePath = path.join(parentPath, "key.txt");

    const key = generateKey();
    fs.writeFileSync(keyFilePath, key);

    processFolder(folderPath, key);
}

main();

File: Kotlin/encryptor.kt
import java.io.File
import java.nio.file.Files
import java.nio.file.Paths
import javax.crypto.Cipher
import javax.crypto.KeyGenerator
import javax.crypto.SecretKey
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec

fun generateKey(): SecretKey {
    val keyGen = KeyGenerator.getInstance("AES")
    keyGen.init(256)
    return keyGen.generateKey()
}

fun padData(data: ByteArray): ByteArray {
    val blockSize = 16
    val paddingLength = blockSize - (data.size % blockSize)
    return data + ByteArray(paddingLength) { paddingLength.toByte() }
}

fun encryptFile(filePath: String, key: SecretKey) {
    val cipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
    val iv = ByteArray(16)
    Files.newInputStream(Paths.get(filePath)).use { it.read(iv) }
    val ivSpec = IvParameterSpec(iv)
    cipher.init(Cipher.ENCRYPT_MODE, key, ivSpec)

    val data = Files.readAllBytes(Paths.get(filePath))
    val paddedData = padData(data)
    val encryptedData = cipher.doFinal(paddedData)

    Files.write(Paths.get(filePath), iv + encryptedData)
}

fun processFolder(folderPath: String, key: SecretKey) {
    File(folderPath).listFiles()?.forEach { file ->
        if (file.isFile) {
            encryptFile(file.path, key)
        }
    }
}

fun main() {
    val folderPath = "E:/Test/Folder"
    val parentPath = File(folderPath).parent
    val keyFilePath = Paths.get(parentPath, "key.txt")

    val key = generateKey()
    Files.write(keyFilePath, key.encoded)

    processFolder(folderPath, key)
}
